core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -5978,7 +5978,7 @@ private void checkNullOperand(SqlTester tester, String op) {
         "^sign('abc')^",
         "Cannot apply 'SIGN' to arguments of type 'SIGN\\(<CHAR\\(3\\)>\\)'\\. Supported form\\(s\\): 'SIGN\\(<NUMERIC>\\)'",
         false);
-    tester.checkType("sign('abc')", "DECIMAL(19, 19) NOT NULL");
+    tester.checkType("sign('abc')", "DECIMAL(19, 0) NOT NULL");
     tester.checkScalar(
```

Comment:
It can only returns `-1`, `1` or `0`, so should the return type always be INTEGER ?

Reply:
You're right, I'll change that.

Reply:
Actually, I think decimal is correct. You are right that it can only return whole numbers but since the sign function returns a datatype matching that of the input, it must return the result as -1.0, 0.0, etc due to how chars are interpreted. I am not fully aware of my chars are evaluated  within the sign function but the return type to this check was set to decimal before I modified this file and after some testing it appears to be appropriate.

Reply:
Thanks for the clarification, we need more tests there.